### DISCUSSION TITLE
Sending events to self

### DISCUSSION THREAD

**Discussion by crystalthoughts** [statelyai/xstate#4930] [General]
Hi, 
I often find myself wanting to orchestrate parallel nested states - things that might otherwise be separate actors but share a common context ( or simply belong next to each other ). 
Is it an anti pattern for an actor to send a message to itself? If so why?

Imagine for example a UI state waiting for a Sync state to send 'synced'. It seems cleaner sometimes to set it up with events rather than create lots of inline promises or separate actor definitions. Also when using the visual debugger one can see how everything works in tandem.

**Comment by davidkpiano** [comment]
It's definitely not an anti-pattern, and it's why `raise(event)` exists! 

📖 https://stately.ai/docs/actions#raise-action

**Comment by crystalthoughts** [comment]
Aha, thanks!
